### ISSUE TITLE
foreground image mask F = Mn ⊗ I,  background image mask B = (1−Mn)⊗In,

### ISSUE THREAD

**Issue by zhending111** [sourachakra/SCoSPARC#1]
Good job!    
I is the image or the vit out feature of the image?Where can i get the supplementary?

**Comment by sourachakra** [comment]
Hello, thank you for bringing this up. 

The updated paper (http://arxiv.org/abs/2403.11107) now reflects the fact that we actually multiply the predicted mask M_n from stage 1 with the ViT patch embeddings (after resizing the M_n to the patch embedding tensor's height and width). We do not mask the image directly with the predicted cross-attention map to obtain the foreground and the background average embeddings (as followed in existing works). 

Also, we added the supplementary material at the end of the main paper - please find in the updated paper link above. Thanks!

**Comment by sourachakra** [comment]
> Good job! I is the image or the vit out feature of the image?Where can i get the supplementary?

Hello, thank you for bringing this up.

The updated paper (http://arxiv.org/abs/2403.11107) now reflects the fact that we actually multiply the predicted mask M_n from stage 1 with the ViT patch embeddings (after resizing the M_n to the patch embedding tensor's height and width). We do not mask the image directly with the predicted cross-attention map to obtain the foreground and the background average embeddings (as followed in existing works).

Also, we added the supplementary material at the end of the main paper - please find in the updated paper link above. Thanks!

**Comment by zhending111** [comment]
> > Good job! I is the image or the vit out feature of the image?Where can i get the supplementary?
> 
> Hello, thank you for bringing this up.
> 
> The updated paper (http://arxiv.org/abs/2403.11107) now reflects the fact that we actually multiply the predicted mask M_n from stage 1 with the ViT patch embeddings (after resizing the M_n to the patch embedding tensor's height and width). We do not mask the image directly with the predicted cross-attention map to obtain the foreground and the background average embeddings (as followed in existing works).
> 
> Also, we added the supplementary material at the end of the main paper - please find in the updated paper link above. Thanks!

Thank you very much for your response. I am very interested in your work and am currently attempting to replicate it. I kindly request your assistance. I have some questions about how the loss function is implemented. I have noticed that the loss changes irregularly and fluctuates greatly according to my code. Below, I have provided a snippet of my early training logs and a portion of the code where the loss is implemented.


code：
def calculate_sal_loss(M, N):
    # M: [N, 1, H, W]
    # N: [N, 1, H, W]

    # Perform element-wise multiplication
    result = M * N

    # Sum over all dimensions
    total_sum = torch.mean(result)

    # Calculate L_sal
    L_sal = 1 - total_sum

    return L_sal


criterion = torch.nn.MSELoss()
def Co_occurenceLoss(object_features,background_features):
    N = object_features.shape[0]
    loss = 0
    for q in range(N):
        for j in range(q, N):

            dij_plus = criterion(object_features[q, :], object_features[j, :])
            dij_minus = criterion(object_features[q, :], background_features[q, :])
            dij_minus = dij_minus + criterion(object_features[j, :], background_features[j, :])
            dij_minus = dij_minus / 2
            loss = loss + torch.log(torch.exp(-dij_plus) / (torch.exp(-dij_plus) + torch.exp(-dij_minus))) * (-1)
    return loss


Epoch [0] -----total loss:25.8941----sal_loss:0.9749-----Lcooc:25.6016 
CoCo  Epoch [0] -----total loss:25.8941----sal_loss:0.9484-----Lcooc:112.0154 
Epoch [0] -----total loss:122.9431----sal_loss:0.9836-----Lcooc:122.6481 
CoCo  Epoch [0] -----total loss:122.9431----sal_loss:0.9489-----Lcooc:96.6496 
Epoch [0] -----total loss:51.1462----sal_loss:0.9554-----Lcooc:50.8596 
CoCo  Epoch [0] -----total loss:51.1462----sal_loss:0.9560-----Lcooc:91.2252 
Epoch [0] -----total loss:29.2736----sal_loss:0.9666-----Lcooc:28.9836 
CoCo  Epoch [0] -----total loss:29.2736----sal_loss:0.9620-----Lcooc:82.2462 
Epoch [0] -----total loss:18.1777----sal_loss:0.9817-----Lcooc:17.8832 
CoCo  Epoch [0] -----total loss:18.1777----sal_loss:0.9688-----Lcooc:81.6811 
Epoch [0] -----total loss:9.0775----sal_loss:0.9639-----Lcooc:8.7883 
CoCo  Epoch [0] -----total loss:9.0775----sal_loss:0.9483-----Lcooc:84.5279 
Epoch [0] -----total loss:28.7361----sal_loss:0.9669-----Lcooc:28.4460 
CoCo  Epoch [0] -----total loss:28.7361----sal_loss:0.9747-----Lcooc:60.8174
